requirements-freqai-rl.txt

```diff
@@ -8,3 +8,6 @@ sb3-contrib==1.7.0; python_version < '3.11'
 # Gym is forced to this version by stable-baselines3.
 setuptools==65.5.1 # Should be removed when gym is fixed.
 gym==0.21; python_version < '3.11'
+# Progress bar for stable-baselines3 and sb3-contrib
+tqdm==4.65.0; python_version < '3.11'
+rich==13.3.3; python_version < '3.11'
```

Comment:
Why both? 

rich does provide a progress bar [on it's own](https://rich.readthedocs.io/en/stable/progress.html) - so this seems pretty much redundant.

If stable baselines requires both, please open an issue / bug report against them (i consider this a bug).

Freqtrade also depends on `progressbar2` - and i REALLY don't see us depending on 3 different progressbar providers - so this will need to change.

Reply:
Unfortunately yes, both libraries are required. it looks like sb3 uses `tqdm`, which uses `rich` to make the progress bar "richer". Here is the Import and ImportError from sb3:

```python
from tqdm.rich import tqdm
...
if tqdm is None:
    raise ImportError(
        "You must install tqdm and rich in order to use the progress bar callback. "
        "It is included if you install stable-baselines with the extra packages: "
        "`pip install stable-baselines3[extra]`"
    )
```
https://tqdm.github.io/docs/rich/
https://github.com/DLR-RM/stable-baselines3/blob/470771b5c2142f0813b6bb1afb9143c960bf09a0/stable_baselines3/common/callbacks.py#L16

Reply:
I don't really see why it wouldn't be possible to use rich directly (see linked documentation above).

Seems like a bug to me.